psydac/fem/grid.py

```diff
@@ -156,6 +145,18 @@ def __init__( self, space, start, end, *, quad_order=None, nderiv=1 , pad=None):
         self._quad_rule_x  = u
         self._quad_rule_w  = w
 
+        #-------------------------------------------
+        # LOCAL GRID, PROPER (WITHOUT GHOST REGIONS)
+        #-------------------------------------------
+
+        # Local indices of first/last elements in proper domain
+        if space.periodic:
+            local_element_start = self._spans.searchsorted( degree + start )
+            local_element_end   = self._spans.searchsorted( degree + end   )
+        else:
+            local_element_start = self._spans.searchsorted( degree if start == 0   else 1 + start)
+            local_element_end   = self._spans.searchsorted( end if end   == n-1 else 1 + end )
```

Comment:
Could you explain why `numpy.searchsorted` should be used here, instead of `list.index`?

Reply:
`numpy.searchsorted` works exactly like `list.index` if the element exist in the list otherwise it will search for the smallest index where the value inserted in it won't break the sorted array.
 when we have repeated knots in a knot sequence, the searched element may not be in the span , so the `list.index` will raise an error of not found element

Reply:
Interesting. I remember making sure that the `span` argument was correct also for repeated knots.

Could you give me a 1D knot sequence example where your fix is necessary?

Reply:
I had this problem with the pipe.h5 file when we run it in parallel,
where we have this knot sequence `[0.    0.    0.    0.125 0.125 0.25  0.25  0.375 0.375 0.5   0.5   0.625
 0.625 0.75  0.75  0.875 0.875 1.    1.    1.   ]`

Reply:
I see, are you running `test_create_geometry_file` in parallel then?

Reply:
I'm talking about the tests in  `test_2d_api_scalar_mapping.py`